include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -1383,10 +1395,28 @@ SpatialOperatorBase<dim, Number>::evaluate_velocity_divergence_term(VectorType &
 
 template<int dim, typename Number>
 void
-SpatialOperatorBase<dim, Number>::update_turbulence_model(VectorType const & velocity)
+SpatialOperatorBase<dim, Number>::update_viscosity(VectorType const & velocity) const
 {
-  // calculate turbulent viscosity locally in each cell and face quadrature point
-  turbulence_model.calculate_turbulent_viscosity(velocity);
+  AssertThrow(param.viscous_problem() and param.viscosity_is_variable(),
+              dealii::ExcMessage(
+                "Updating viscosity reasonable for variable viscosity models only."));
+
+  // reset the viscosity stored
+  // viscosity = viscosity_newtonian_limit
+  set_constant_viscosity(viscous_kernel_data.viscosity);
+
+  // add contribution from generalized Newtonian model
+  // viscosity += generalized_newtonian_viscosity(viscosity_newtonian_limit)
+  if(param.generalized_newtonian_model_data.is_active)
+    generalized_newtonian_model.add_viscosity(velocity);
+
+  // add contribution from turbulence model
+  // viscosity += turbulent_viscosity(viscosity)
+  // note that the apparent viscosity is used to
+  // compute the turbulent viscosity, such that the
+  // *sequence of calls matters*
```

Comment:
for clarity, please add ", i.e. we can only compute the turbulent viscosity once the laminar viscosity has been computed.".

As a minor comment, I would generally let clang-format do the line breaks for comments.